apiserver/application/application.go

```diff
@@ -888,6 +883,14 @@ func (api *API) AddRelation(args params.AddRelation) (params.AddRelationResults,
 		if relName != "" {
 			endpoints[i] = remoteApp.Name() + ":" + relName
 		}
+		isRemote = true
+	}
+	// If it's not a remote relation to another model then
+	// the user needs write access to the model.
+	if !isRemote {
```

Comment:
This is for the non CMR case. In that case, *both* applications are in the same model. So the user needs write access to that one model to relate applications.